ERCS/erc-7860.md

```diff
@@ -0,0 +1,207 @@
+4. Prompts and other data can be encrypted for privacy while remaining verifiable for authenticity.
+   
+## **Specification**
+
+The extension introduces the following interface:
+
+```
+/**
+ * @title IAgentNFT
+ * @dev Interface for ERC721 AgentNFT extension that adds prompt and agent management capabilities
+ * This extension allows NFTs to store and manage prompts and AI Agent's corresponding address
+ */
+interface IAgentNFT  /* is IERC165, IERC721 */ {
+    /**
+     * @dev Emitted when an address is updated for a AgentNFT
+     * @param tokenId The ID of the AgentNFT
+     * @param agent The new agent address
+     */
+    event AgentUpdated(uint256 indexed tokenId, address indexed agent);
+
+    /**
+     * @dev Emitted when a prompt is updated for a AgentNFT
+     * @param tokenId The ID of the AgentNFT
+     * @param prompt The new prompt string
+     */
+    event PromptUpdated(uint256 indexed tokenId, string prompt);
+
+    /**
+     * @dev Returns the current address corresponding to the AgentNFT. The address can be a wallet address or smart contract address. The AI Agent is represented by the AgentNFT of the given tokenId parameter
+     * @param tokenId The ID of the AgentNFT to query
+     * @return The address of the associated AI agent
+     */
+    function agent(uint256 tokenId) external view returns (address);
+
+    /**
+     * @dev Updates the address for an AgentNFT
+     * @param tokenId The ID of the AgentNFT to update
+     * @param newAgent The new agent address to assign
+     */
+    function updateAgent(uint256 tokenId, address newAgent) external;
+
+    /**
+     * @dev Returns the current prompt for a AgentNFT
+     * @param tokenId The ID of the AgentNFT to query
+     * @return The prompt string associated with the AgentNFT
+     */
+    function prompt(uint256 tokenId) external view returns (string memory);
+
+    /**
+     * @dev Updates the prompt for an AgentNFT
+     * @param tokenId The ID of the AgentNFT to update
+     * @param newPrompt The new prompt string to assign
+     */
+    function updatePrompt(uint256 tokenId, string memory newPrompt) external;
+}
+
+```
+
+## **Rationale**
+
+This interface is designed to be minimal yet flexible, ensuring compatibility with various use cases:
+
+1. On-Chain Agent Prompts – Prompts are a core asset of AI agents. By storing them within AgentNFTs, they become transferable, preserving their value and utility.
+2. Agent Information Authentication – On-chain information can be used for identification and authentication, enhancing transparency and trust.
+3. Efficient State Querying – The interface enables seamless tracking and updating of agent information, facilitating efficient state management.
+
+## **Backwards Compatibility**
+
+This extension is fully backwards compatible with ERC-721. 
+
+## **Reference Implementation**
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code within the document itself must be CC0-1.0

You can include MIT code in the assets directory and link to it.